solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRQuery.java

```diff
@@ -77,7 +78,7 @@ public RankQuery wrap(Query _mainQuery) {
     if (_mainQuery != null) {
       mainQuery = _mainQuery;
     }
-    reRankModel.setOriginalQuery(mainQuery);
+    reRankRescorer.setOriginalQuery(mainQuery);
```

Comment:
I agree that both the LTRCollector and the LTRWeight shouldn't have to create a new LTRRescorer on their own, it should be passed in instead.  However, I think you pulled the rescorer up too far.
The Rescorer is an implementation detail on how the LTRQuery will perform the reranking, and the LTRQParser should not have to know anything about it.  The LTRQuery should continue to take in a ModelQuery in the constructor, and internally the LTRQuery can construct one.  Also, with this you do not have to re-implement hashCode, equals, toString, and setOriginalQuery in the LTRRescorer, which just wraps the ModelQuery calls.  Since the LTRQuery will still take in a ModelQuery, it can use it like it did before for those functions instead.  This way you can also just keep around a Rescorer reference in LTRQuery instead of an LTRRescorer reference without the need for those specialized functions anymore.